File: C++/659.cpp
#include <iostream>

using namespace std;

// 659 - Números Jeroglíficos - Golgotha's Melancholy - https://github.com/Golgotha-sMelancholy

int main()
{
    int number = 0;
    cin >> number;
    while (number != 0) {
        string hieroglyphic = "";
        while (number >= 1000000) {
                hieroglyphic += 'H';
                number -= 1000000;
            }
            while (number >= 100000) {
                hieroglyphic += 'R';
                number -= 100000;
            }
            while (number >= 10000) {
                hieroglyphic += 'D';
                number -= 10000;
            }
            while (number >= 1000) {
                hieroglyphic += 'F';
                number -= 1000;
            }
            while (number >= 100) {
                hieroglyphic += 'C';
                number -= 100;
            }
            while (number >= 10) {
                hieroglyphic += 'G';
                number -= 10;
            }
            while (number >= 1) {
                hieroglyphic += 'T';
                number -= 1;
            }
        cout << hieroglyphic << endl;
        cin >> number;
    }
}

File: Java/659.java
import java.util.Scanner;

/**
 *
 * 659 - Números Jeroglíficos - Golgotha's Melancholy - https://github.com/Golgotha-sMelancholy
 */
public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int numero = sc.nextInt();
        while (numero > 0) {
            String jeroglifico = "";
            while (numero >= 1000000) {
                jeroglifico += 'H';
                numero -= 1000000;
            }
            while (numero >= 100000) {
                jeroglifico += 'R';
                numero -= 100000;
            }
            while (numero >= 10000) {
                jeroglifico += 'D';
                numero -= 10000;
            }
            while (numero >= 1000) {
                jeroglifico += 'F';
                numero -= 1000;
            }
            while (numero >= 100) {
                jeroglifico += 'C';
                numero -= 100;
            }
            while (numero >= 10) {
                jeroglifico += 'G';
                numero -= 10;
            }
            while (numero >= 1) {
                jeroglifico += 'T';
                numero -= 1;
            }
            System.out.println(jeroglifico);
            numero = sc.nextInt();
        }

    }

}
